### ISSUE TITLE
 Agent gets Unhealthy on reassigning multiple times to different policies.

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/elastic-agent#1665]
**Kibana version:** 8.6 SNAPSHOT Kibana cloud environment

**Host OS:** Ubuntu 20, Rhel 9

**Build details:**
VERSION: 8.6 SNAPSHOT
BUILD: 57815
COMMIT: 7a3243b79fbc309875490e1477a3d97b80854676

**Preconditions:**
1. 8.6 SNAPSHOT Kibana cloud environment should be available.
2. Multiple agents should be installed using Agent Policy having System and Endpoint integration.
3. Multiple other Agent policy should be created having System integration.

**Steps to reproduce:**
1. Navigate to Agents tab.
2. Assign agent to new policy having System integration.
3. Observe agent in updating state and then moves back to Healthy state.
4. Similarly, on multiple re-assigning observe agent goes to Unhealthy state.

**Screenshots:**
![2](https://user-images.githubusercontent.com/77374876/199178479-246e3ec5-ff6a-41c5-8a71-542c48ffdf53.JPG)
![5](https://user-images.githubusercontent.com/77374876/199178495-d5378991-b717-465c-9a51-ff040e01bd85.JPG)

**What else not working:**
- Agent gets unhealthy inconsistently on adding Endpoint Security integration.[Observed for Rhel9 agent]
![3](https://user-images.githubusercontent.com/77374876/199178490-4b925e75-1cd3-424e-8929-3673a47ff9dc.JPG)

**Agent Logs:**
[elastic-agent-diagnostics-2022-11-01T06-11-46Z-00.zip](https://github.com/elastic/elastic-agent/files/9907864/elastic-agent-diagnostics-2022-11-01T06-11-46Z-00.zip)

**Expected Result:**
 Agents should not get Unhealthy on reassigning multiple times to different policies.

**Comment by amolnater-qasource** [comment]
@manishgupta-qasource Please review.

**Comment by manishgupta-qasource** [comment]
Secondary review for this ticket is Done

**Comment by cmacknz** [comment]
The `Error creating runner from config: Can only start an input when all related states are finished` is the only notable problem I can see in the logs.

```json
{"log.level":"error","@timestamp":"2022-11-01T06:10:37.789Z","log.origin":{"file.name":"log/reporter.go","file.line":36},"message":"2022-11-01T02:10:37-04:00 - message: Application: filebeat--8.6.0-SNAPSHOT[71bbf1a4-d4d2-42b0-87a0-99b860834f4a]: State changed to FAILED: 1 error occurred:\n\t* 1 error: Error creating runner from config: Can only start an input when all related states are finished: {Id: native::37189-64768, Finished: false, Fileinfo: &{messages 119742 384 {830344045 63802879097 0x55e2a038d240} {64768 37189 1 33152 0 0 0 0 119742 4096 256 {1667282255 62451952} {1667282297 830344045} {1667282297 830344045} [0 0 0]}}, Source: /var/log/messages, Offset: 127294, Timestamp: 2022-11-01 02:10:37.364184481 -0400 EDT m=+742.423814143, TTL: -1ns, Type: log, Meta: map[], FileStateOS: 37189-64768}\n\n - type: 'ERROR' - sub_type: 'FAILED'","ecs.version":"1.6.0"}
```

@belimawr or @rdner do you recognize this error? Any ideas what can cause it?

**Comment by rdner** [comment]
@cmacknz I see that the same error message has been reported before in https://github.com/elastic/beats/issues/31929 and it was supposed to be fixed by https://github.com/elastic/beats/pull/32309 

@belimawr should be able to give more details since he was working on it.

**Comment by belimawr** [comment]
No idea aside the PR @rdner mentioned. I'll try reproducing it today.

**Comment by belimawr** [comment]
I managed to reproduce the issue, as @rdner said, it seems to be the same as https://github.com/elastic/beats/issues/31929 and it should have been fixed. I'm still investigating to understand why the bug is still there.

To reproduce it you only need one Elastic-Agent. But it depends on the host, on a Archlinux VM it did not occur, but on Ubuntu 20.04 it was easily reproducible. It seems relates with some of the system log files the system integration reads that seem to differ on Archlinux.

It can also be reproduced on a standalone Elastic-Agent by copying different policies onto the `elastic-agent.yml`, Elastic-Agent will reload the file, which has got pretty much the same effect as changing a policy on Kibana for a Fleet managed Elastic-Agent.
